vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -128,61 +128,68 @@ def _update_nested_graph_properties(
+
+    # It's not possible to address nested argument of data_frame like data_frame.x.y, just top-level ones like
+    # data_frame.x.
+    config["data_frame"] = {}
+
+    for ctd in ctd_parameters:
+        selector_value = ctd[
+            "value"
+        ]  # TODO: needs to be refactored so that it is independent of implementation details
+        if hasattr(selector_value, "__iter__") and ALL_OPTION in selector_value:  # type: ignore[operator]
+            selector: SelectorType = model_manager[ctd["id"]]
+            selector_value = selector.options
+        selector_value = _validate_selector_value_none(selector_value)
+        selector_actions = _get_component_actions(model_manager[ctd["id"]])
 
-                if action.function._function.__name__ != "_parameter" or target not in action_targets:
-                    continue
+        for action in selector_actions:
+            action_targets = _create_target_arg_mapping(action.function["targets"])
 
-                for action_targets_arg in action_targets[target]:
-                    graph_config = _update_nested_graph_properties(
-                        graph_config=graph_config, dot_separated_string=action_targets_arg, value=selector_value
-                    )
+            if action.function._function.__name__ != "_parameter" or target not in action_targets:
+                continue
 
-        parameterized_config[target] = graph_config
+            for action_targets_arg in action_targets[target]:
+                config = _update_nested_graph_properties(
+                    graph_config=config, dot_separated_string=action_targets_arg, value=selector_value
+                )
 
-    return parameterized_config
+    return config
 
 
 # Helper functions used in pre-defined actions ----
-def _get_filtered_data(
-    targets: List[ModelID],
-    ctds_filters: List[CallbackTriggerDict],
+def _get_targets_data_and_config(
+    ctds_filter: List[CallbackTriggerDict],
     ctds_filter_interaction: List[Dict[str, CallbackTriggerDict]],
-) -> Dict[ModelID, pd.DataFrame]:
-    filtered_data = {}
+    ctds_parameters: List[CallbackTriggerDict],
+    targets: List[ModelID],
+):
+    all_filtered_data = {}
+    all_parameterized_config = {}
+
     for target in targets:
+        # parametrized_config includes a key "data_frame" that is used in the data loading function.
+        parameterized_config = _get_parametrized_config(target=target, ctd_parameters=ctds_parameters)
         data_source_name = model_manager[target]["data_frame"]
-        data_frame = data_manager[data_source_name].load()
-        data_frame = _apply_filters(data_frame=data_frame, ctds_filters=ctds_filters, target=target)
-        data_frame = _apply_filter_interaction(
-            data_frame=data_frame, ctds_filter_interaction=ctds_filter_interaction, target=target
+        data_frame = data_manager[data_source_name].load(**parameterized_config["data_frame"])
```

Comment:
@maxschulz-COL most of the changes in this file are just refactoring. This line is the only "real" change where we pass `parameterized_config["data_frame"]` through to the `load` function.